frontend/public/components/overview/namespace-overview.tsx

```diff
@@ -44,6 +45,31 @@ const OverviewNamespaceSummary = ({ns}) => <div className="group">
   </div>
 </div>;
 
+const OverviewLinks = ({ns, consoleLinks}) => {
+  const namespaceDashboardLinks = _.sortBy(_.filter(consoleLinks, link => link.spec.location === 'NamespaceDashboard'), 'spec.text');
+  const links = _.compact(_.map(namespaceDashboardLinks, link => {
+    const namespacesToDisplayLink = _.get(link, ['spec', 'namespaceDashboard', 'namespaces']);
+    // if the link is restricted to specific a namespace or namespaces, only display the link if the current namespace matches
+    if (namespacesToDisplayLink && !_.includes(namespacesToDisplayLink, ns.metadata.name)) {
+      return;
+    }
+    return <li key={link.metadata.uid}><ExternalLink href={link.spec.href} text={link.spec.text} /></li>;
+  }));
```

Comment:
I'd filter once for both link location and the namespaces. That way you can skip `_.compact` and only sort the links you'll show. Something like...

```js
const filtered = _.filter(consoleLinks, (link: K8sResourceKind) => {
  if (link.spec.location !== 'NamespaceDashboard') {
    return false;
  }
  const namespaces = _.get(link, 'spec.namespaceDashboard.namespaces');
  return !namespaces || _.includes(namespaces, ns.metadata.name);
});

const dashboardLinks = _.sortBy(filtered, 'spec.text').map((link: K8sResourceKind) => {
  return <li key={link.metadata.uid}><ExternalLink href={link.spec.href} text={link.spec.text} /></li>
});
```

Reply:
Thanks!  I've refactored the code a bit. I had started down this path prior to your comment, so it looks like you're getting me trained.  ;-)